tools/lsp/common/rename_component.rs

```diff
@@ -0,0 +1,781 @@
+                result.push(
+                    specifier
+                        .ExportName()
+                        .map(|n| n.text().to_string())
+                        .unwrap_or_else(|| type_name.to_string()),
+                );
+            }
+        }
+        if let Some(component) = export.Component() {
+            if component.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+        for structs in export.StructDeclaration() {
+            if structs.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+        for enums in export.EnumDeclaration() {
+            if enums.DeclaredIdentifier().text() == type_name {
+                result.push(type_name.to_string());
+            }
+        }
+    }
+
+    result.sort();
+    result
+}
+
+fn replace_element_types(
+    element: &syntax_nodes::Element,
+    old_type: &str,
+    new_type: &str,
+    edits: &mut Vec<(SourceFile, lsp_types::TextEdit)>,
+) {
+    if let Some(name) = element.QualifiedName() {
+        if name.text().to_string().trim() == old_type {
+            edits.push((
+                element.source_file.clone(),
+                lsp_types::TextEdit {
+                    range: util::map_node(&name).expect("Just had a source_file"),
+                    new_text: new_type.to_string(),
+                },
+            ))
+        }
+    }
+
+    for c in element.children() {
+        match c.kind() {
+            SyntaxKind::SubElement => {
+                let e: syntax_nodes::SubElement = c.into();
+                replace_element_types(&e.Element(), old_type, new_type, edits);
+            }
+            SyntaxKind::RepeatedElement => {
+                let e: syntax_nodes::RepeatedElement = c.into();
+                replace_element_types(&e.SubElement().Element(), old_type, new_type, edits);
+            }
+            SyntaxKind::ConditionalElement => {
+                let e: syntax_nodes::ConditionalElement = c.into();
+                replace_element_types(&e.SubElement().Element(), old_type, new_type, edits);
+            }
+            _ => { /* do nothing */ }
+        }
+    }
+}
+
+pub fn fix_imports(
+    type_loader: &TypeLoader,
+    exporter_path: &Path,
+    old_type: &str,
+    new_type: &str,
+    edits: &mut Vec<(SourceFile, lsp_types::TextEdit)>,
+) {
+    for doc in type_loader.all_documents() {
+        let Some(doc_path) =
+            doc.node.as_ref().and_then(|n| n.source_file()).map(|sf| sf.path().to_owned())
+        else {
+            continue;
+        };
+        if doc_path.starts_with("builtin:") {
+            continue;
+        }
+        if &doc_path == exporter_path {
+            continue;
+        }
+        fix_import_in_document(type_loader, doc, &exporter_path, old_type, new_type, edits);
+    }
+}
+
+pub fn fix_import_in_document(
```

Comment:
```suggestion
fn fix_import_in_document(
```

It's nice to keep some hygiene even for internal modules. If a function is not used outside of the module, keep it private. That way this also communicate that this is an internal function that can be changed when refactoring. 

Also valid for other functions in this file.